src/main/java/com/marek/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -29,15 +29,15 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
     @Override
     public ShoppingCartDto findUserCart(User user) {
         return shoppingCartMapper.toDto(shoppingCartRepository
-                .findShoppingCartByUser(user));
+                .findShoppingCartByUser(user.getId()));
```

Comment:
You are getting the user's id from the user and passing it to the findShoppingCartByUser method. If the user is null, this will cause a NullPointerException. Consider adding a null check for the user before using it.